features/mbedtls/targets/TARGET_STM/md5_alt.c

```diff
@@ -175,6 +175,12 @@ void mbedtls_md5_finish( mbedtls_md5_context *ctx, unsigned char output[16] )
             return; // Return error code here
         }
     }
+    /* The following test can happen when the input is empty, and mbedtls_md5_update has never been called */
```

Comment:
@adustm  
1. This seems to introduce a redundancy with the pre-existing code https://github.com/ARMmbed/mbed-os/pull/5630/files#diff-00438dce44385c38da7359547815b2daR135 handling the case of an `update` call with an empty buffer.
2. Why are we not unconditionally calling `HAL_HASH_MD5_Accumulate(&ctx->hhash_md5, ctx->sbuf, ctx->sbuf_len) ` here, even if `ctx->sbuf_len == 0`? Looking at the [code](https://github.com/ARMmbed/mbed-os/blob/master/targets/TARGET_STM/TARGET_STM32F7/device/stm32f7xx_hal_hash.c#L622) for `HAL_HASH_MD5_Accumulate`, it always performs the same check for `hhash_md5.Phase` and updates `HASH->CR` if it finds  `HAL_HASH_PHASE_READY`. Is there any problem with calling `HAL_HASH_MD5_Accumulate` with an empty buffer?

Reply:
Hello,
Thank you for this finding. 
I've tested your suggested modification and it works fine. You can find it in the next commit.
Kind regards